### PR TITLE
add disabled option to ListItem

### PR DESCRIPTION
**Pull request by eszthoff**
[JF-1459](https://jira.textkernel.nl/browse/JF-1459) Adds disabled option to ListItem where item is not highlighted on hover

### CODE REVIEW

File: src/components/List/ListItem/ListItem.scss [hunk 1]
```diff
@@ -18,4 +18,8 @@
     &--clickable {
         cursor: pointer;
     }
+
+    &--disabled:hover {
+        background-color: var(--color-background);
```
[the remark this answers is not in the document]
Comment (SebastiaanPasterkamp) [reply src/components/List/ListItem/ListItem.scss]: Maybe add a `--color-muted-background: #{mix($color-background, $color-muted, 90)};` to the colors, and apply it to the `&--disabled {` style (without `:hover`)?

File: src/components/List/ListItem/ListItem.js [hunk 2]
```diff
@@ -31,16 +31,19 @@ ListItem.displayName = 'ListItem';
...
-    onClick: PropTypes.func
+    /** Format this item as disabled */
+    disabled: PropTypes.bool
 };
 
 ListItem.defaultProps = {
     children: null,
     isSelected: false,
-    onClick: null
+    onClick: null,
```
[the remark this answers is not in the document]
Comment (eszthoff) [reply src/components/List/ListItem/ListItem.js]: haha. OCD ;-)

File: src/components/List/ListItem/ListItem.js [hunk 3]
```diff
@@ -7,11 +7,11 @@ import styles from './ListItem.scss';
 const { block, elem } = bem({
     name: 'ListItem',
     classnames: styles,
-    propsToMods: ['isSelected', 'onClick']
+    propsToMods: ['isSelected', 'onClick', 'disabled']
 });
 
 const ListItem = props => {
-    const { children, isSelected, onClick, ...rest } = props;
+    const { children, isSelected, onClick, disabled, ...rest } = props;
```
[the remark this answers is not in the document]
Comment (eszthoff) [reply src/components/List/ListItem/ListItem.js]: Hopefully not. But that is for the one who implements it, no? Adding tests for it, just makes it more complicated, with little benefits

File: src/components/List/ListItem/ListItem.scss (hunk 1 above)
[the remark this answers is not in the document]
Comment (eszthoff) [reply src/components/List/ListItem/ListItem.scss]: The point is that you cannot see any changes when it is disabled. It is defined here to background to contradict the rule above during hover.

File: src/components/List/ListItem/ListItem.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (eszthoff) [reply src/components/List/ListItem/ListItem.js]: fixed

File: src/components/List/ListItem/ListItem.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (SebastiaanPasterkamp) [reply src/components/List/ListItem/ListItem.js]: Okay. Indeed any implementation can opt to pass `onClick` or not. Just checking.

File: src/components/List/ListItem/ListItem.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (SebastiaanPasterkamp) [reply src/components/List/ListItem/ListItem.js]: :smile:

File: src/components/List/ListItem/ListItem.scss (hunk 1 above)
[the remark this answers is not in the document]
Comment (SebastiaanPasterkamp) [reply src/components/List/ListItem/ListItem.scss]: Perhaps `&:hover:not(&--disabled)`?

**Pull request merged**